ctru-rs/src/console.rs

```diff
@@ -58,12 +58,12 @@ pub enum Dimension {
 /// you can try using [`Soc::redirect_to_3dslink`](crate::services::soc::Soc::redirect_to_3dslink) while activating the `--server` flag for `3dslink` (also supported by `cargo-3ds`).
 /// More info in the [`cargo-3ds` docs](https://github.com/rust3ds/cargo-3ds#running-executables).
 #[doc(alias = "PrintConsole")]
-pub struct Console<'screen> {
+pub struct Console<'screen, S: Screen> {
     context: Box<PrintConsole>,
-    screen: RefMut<'screen, dyn Screen>,
+    screen: RefMut<'screen, S>,
 }
```

Comment:
I seem to remember some long discussion about why this ended up as `RefMut<dyn Screen>` rather than something like the `Console<'s, S>` you have here... I don't remember all the reasoning or necessarily have a strong opinion, but I think it should still be possible to use `dyn` here if we want, with something like this:

```rs
pub struct Console<'screen> {
    context: Box<PrintConsole>,
    screen: RefMut<'screen, dyn FlushScreen>,
}

trait FlushScreen: Flush + Swap + Screen {}
impl<T: Flush + Swap + Screen> FlushScreen for T {}
```

The only caveat I can think of might be that `new()` would then require either `RefMut<dyn FlushScreen>`, or something like `pub fn new<S: Flush + Screen + Swap>(screen: RefMut<'screen, S>)` and convert the concrete type with `RefMut::map`.

I'm not really sure what is simpler and I don't have a strong preference, although keeping it as `dyn` would probably make https://github.com/rust3ds/test-runner/pull/15 unnecessary, at least.

Reply:
Thanks, this is a better approach. The only downside is that it prevents us from using TopScreenLeft/Right as the screen type, but from looking at the libctru code, it implicitly expects the screen side to be left (from `consoleInit`: `console->frameBuffer = (u16*)gfxGetFramebuffer(screen, GFX_LEFT, NULL, NULL);`).

Reply:
yeah, AFAIK the console is not intended to work with left/right or anything (basically just wide mode or standard). I think at one point I asked the devkitPro folks about supporting buffering / more advanced rendering (e.g. for TUI roguelikes or something), and they were not interested in supporting those kinds of use cases (which prompted me to try myself with https://github.com/ian-h-chamberlain/console_3ds although I have since stopped working on it).